apps/site/__tests__/quiz.spec.tsx

```diff
@@ -89,4 +90,57 @@ describe("QuizHeader", () => {
 			}
 		});
 	});
+
+	it("should open the success dialog", () => {
+		mockUseQuiz.mockReturnValue({ evaluations: {}, completionState: "completed" });
+
+		render(
+			<QuizHeader
+				lesson={mockLesson}
+				course={mockCourse}
+				questions={mockQuestions}
+				currentIndex={0}
+				goToQuestion={jest.fn()}
+			/>
+		);
+
+		const successDialog = screen.getByText("Geschafft!");
+		expect(successDialog).toBeInTheDocument();
+	});
+
+	it("should open the failed dialog", () => {
+		mockUseQuiz.mockReturnValue({ evaluations: {}, completionState: "failed" });
+
+		render(
+			<QuizHeader
+				lesson={mockLesson}
+				course={mockCourse}
+				questions={mockQuestions}
+				currentIndex={0}
+				goToQuestion={jest.fn()}
+			/>
+		);
+
+		const failDialog = screen.getByText("Nicht Bestanden");
+		expect(failDialog).toBeInTheDocument();
+	});
+
+	it("should not open any dialog", () => {
```

Comment:
die testfälle sollten auich das setup beinhalten. 

zB "It should not open completion-dialog when completionState is in progress"

Also auch die conditions. Das gilt für alle Testfälle.

Ansonsten gute erste tests!